File: 1-two-sum/two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> numDict = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++) {
            int complement = target - nums[i];

            if (numDict.ContainsKey(complement)) {
                return new int[] { numDict[complement], i };
            }

            numDict[nums[i]] = i;
        }

        // If no solution is found, you may throw an exception or return null based on your requirements
        throw new ArgumentException("No solution found");
    }
}

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const mapIndex = [];
    const mapVal = []
    for(let i = 0; i < nums.length;i++){
        const cur = nums[i];
        const x = target - cur;
        if(mapVal.includes(x)){
            return [i,mapVal.indexOf(x)]
        }
        mapIndex.push(i)
        mapVal.push(cur)
    }
};